### ISSUE TITLE
Homogenous Profile underdocumented

### ISSUE THREAD

**Issue by s9105947** [ComputationalRadiationPhysics/picongpu#4020]
Dear Maintainers,

the `Homogenous` profile is never explained, [the respective section](https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/include/picongpu/param/density.param#L90) is as follows:

```c++
/* definition of homogenous profile */
using Homogenous = HomogenousImpl;
```

This is not clear (at least to me).
I'd like to suggest a more detailed explanation.

*Note: The improved documentation could also explicitly mention that the [traditionally correct spelling](https://www.lexico.com/en/definition/homogeneous) "homogeneous" is never used within the source.*

**Comment by sbastrakov** [comment]
Hello @s9105947 ,

This instance technically follows the same scheme as other profiles (like `using Gaussian = GaussianImpl<GaussianParam>;` just above), and unlike them `HomogenousImpl` is just a struct, not a template.

I agree with lack of comments in this part, it applies to all such density profiles, and those comments should be placed in respective `.def` files i think. E.g. all of them may prompt a user's question "so this is profile, but with which density and number of macroparticles per cell?", which is answered [in another place](https://picongpu.readthedocs.io/en/latest/usage/param/particles/init.html#createdensity). I will try to better link these pieces together.

Regarding the naming, i think this one is quite safe to rename, since it's not widely used anyways so is not likely to crash a lot of existing setups

**Comment by sbastrakov** [comment]
Now some description is added at start of `density.param` which is also included [here](https://picongpu.readthedocs.io/en/latest/usage/param/core.html#density-param). Doxygen comments are also added to declarations of classes/templates like `HomogenousImpl`, how easily are they hooked up depends on one's development environment.

Regarding the renaming part, I see no problem with doing it and so can do it now, unless there are objections.

**Comment by s9105947** [comment]
Thank you for adding the comments.

Is this explanation clear enough? Especially in terms of "what is the resulting density", i.e. "what is the density in SI units"?
(I assume it refers to setting BASE_DENSITY for the entire simulation volume.)

Renaming is probably overkill, because 1) that's a breaking change with not too much benefit (either spelling is complicated) and 2) the better alternative would be sth like "constant" or "uniform" -- which is an even stronger break.
(The spelling is now mentioned in this issue anyways, this should be enough documentation for such a minor detail.)

**Comment by sbastrakov** [comment]
>Is this explanation clear enough? Especially in terms of "what is the resulting density", i.e. "what is the density in SI units"?
(I assume it refers to setting BASE_DENSITY for the entire simulation volume.)

I am not sure, probably I am also too used to the existing system to judge. The logic is that a user sets `BASE_DENSITY_SI` as normalization unit for the whole simulation, in particles/m^3 (it's also explained in the comments to that variable). And then whatever profile is used by default operates with this unit as the basis, so when profile returns 2 it means the density is `2 * BASE_DENSITY_SI` in SI or similarly `2 * internal_density_unit` in our internal unit systems. This default can be overridden per species using `densityRatio`, but then the assumption is a user knows what they are doing.

**Comment by s9105947** [comment]
To be fair, this is not mentioned for the other profiles either, only for the `BASE_DENSITY_SI` as you've said.

When I tried to use the homogeneous profile at first I thought "well and how do I set the actual density I want everwhere" until it dawned on me that I am supposed to use `BASE_DENSITY_SI` and `densityRatio` to express that, and that the homogeneous is simply always `1`.

I am not sure if this is supposed to go into a profile docstring or into a more general guide in the documentation. (Pet Peeve: "Placing Macroparticles"/"Creating a Density" deserves its own first-level subheading. Unfortunately I don't have enough time on my hands to write this section myself.)

The current version is consistent with the other profiles/the rest of PIConGPU -> leave it be?
